fea-rs/src/compile/lookups.rs

```diff
@@ -49,9 +49,12 @@ use gsub_builders::{
 };
 pub(crate) use helpers::ClassDefBuilder2;
 
-pub trait Builder {
+// A simple trait for building lookups
+// This exists because we use it to implement `LookupBuilder<T>`
+pub(crate) trait Builder {
     type Output;
-    fn build(self) -> Self::Output;
+    // the var_store is only used in GPOS, but we pass it everywhere
+    fn build(self, var_store: &mut VariationStoreBuilder) -> Self::Output;
```

Comment:
agree this is annoying, and unfortunately `Option<&mut T>` is annoying to work with, because it doesn't implement `Copy`, which means you need to do a dance to get the inner reference anywhere you need it, so I found this to be the best compromise.

I agree that this is confusing, though, and I will expand the comment to explain what's going on.